Resources/NetHookAnalyzer2/NetHookAnalyzer2/NetHookDump.cs

```diff
@@ -23,26 +26,34 @@ public void LoadFromDirectory(string directory)
 			var itemFiles = directoryInfo.EnumerateFiles("*.bin", SearchOption.TopDirectoryOnly);
 			foreach (var itemFile in itemFiles)
 			{
-				var item = new NetHookItem();
-				if (item.LoadFromFile(itemFile))
-				{
-					items.Add(item);
-				}
+				AddItemFromFile(itemFile);
 			}
 		}
 
 		public IEnumerable<NetHookItem> Items => readOnlyView;
 
-		public NetHookItem AddItemFromPath(string path)
+		public byte[] GetAccountAuthSecrets(int secretId) => accountAuthSecrets.GetValueOrDefault(secretId);
+
+		public NetHookItem AddItemFromFile(FileInfo fileInfo)
 		{
-			var fileInfo = new FileInfo(path);
 			var item = new NetHookItem();
 			if (!item.LoadFromFile(fileInfo))
 			{
 				return null;
 			}
 
 			items.Add(item);
+
+			if (item.EMsg == SteamKit2.EMsg.ServiceMethodResponse && item.InnerMessageName == "Credentials.GetAccountAuthSecret#1")
```

Comment:
So this relies upon us parsing the dump in chronological order?

Reply:
When it loads this secrets payload, earlier dumps will also decrypt because the decryption happens when you click on the item. But you do need to have this payload to be dumped at some point for this to work.

Reply:
Should we require the payload to be in the _same_ dump, or could we make `accountAuthSecrets` the static object instead?